### PR TITLE
added Grad-CAM implementation

### PR DESCRIPTION
**Pull request by ruthcfong**
Here's an implementation of Grad-CAM that extends SaliencyMask class.

I've included a few optional arguments in the GetMask function (i.e., should_resize, three_dims) to allow the option to return the original, low-res 2D Grad-CAM heatmap, which I need.

### CODE REVIEW

**Comment by ruthcfong** [comment]
I signed it!

File: grad_cam.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to compute SaliencyMasks."""
+import numpy as np
+import tensorflow as tf
+
+from saliency import SaliencyMask
+from skimage.transform import resize
```
Comment (dsmilkov) [inline grad_cam.py]: Since the core library is trying to avoid dependencies, how about using tf.image.resize_bilinear(...) instead of skimage

**Comment by ruthcfong** [comment]
I've replaced skimage with tf's resize function as suggested and updated the README file.

**Comment by nsthorat** [comment]
+1 thanks for the contribution! :)

**Comment by nalinimsingh** [comment]
@ruthcfong -- I'm curious about [this line](https://github.com/PAIR-code/saliency/blob/ba88a40cd2ccb1a0c1390fd85d569f9a89ef8d3c/saliency/grad_cam.py#L64) in the GradCAM implementation here. 

Why is this matrix initialized with ones -- from my read of the paper, it seems it should be filled with zeroes? 

[This doesn't affect the qualitative trend you see across the image in the GradCAM masks I've generated and looked at since this is a constant shift at each pixel, but it does slightly perturb some of the numeric values.]

**Comment by ruthcfong** [comment]
@nalinimsingh unfortunately, I don't remember what my reasoning was when I added the GradCAM implementation. I think you're right though that the proper way to do it is to initialize with zeros.

My hypothesis that I probably was looking at this keras implementation by jacobgil (committed March 2017; my commit was July 2017), which does the [same thing](https://github.com/jacobgil/keras-grad-cam/blob/master/grad-cam.py#L106).

**Pull request merged**